### PR TITLE
Develop: Add more integration tests to the Patient and Person service #902

### PR DESCRIPTION
**Pull request by josephbate**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [ ] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary
hello everyone 
i decided to use the dbunit library aproach to do the intergration tests
i have created a `DbUnitUtility.java` and a dataset called `personDataSet.xml` then called it out in the `PersonServiceTest.java`is the the data is the db that is know the number of person registered in the dataBase 
i creat a test to test 
unfortunatenely i have encounter a setback the data is not loading in the active data base at runtime simply because postgres is not running at the port 5432 which got from `PostgreSQLContainer`; ` public static final Integer POSTGRESQL_PORT = 5432;`
any logical ideas are welcome

## error 
`org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections. at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:342) at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:54) at org.postgresql.jdbc.PgConnection.(PgConnection.java:263) at org.postgresql.Driver.makeConnection(Driver.java:443) at org.postgresql.Driver.connect(Driver.java:297) at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677) at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228) at org.dbunit.JdbcDatabaseTester.getConnection(JdbcDatabaseTester.java:116) at org.dbunit.AbstractDatabaseTester.executeOperation(AbstractDatabaseTester.java:194) at org.dbunit.AbstractDatabaseTester.onSetup(AbstractDatabaseTester.java:109) at org.openelisglobal.person.PersonServiceTest.init(PersonServiceTest.java:38) at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75) at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86) at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84) at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:251) at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97) at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)Caused by: java.net.ConnectException: Connection refused (Connection refused) at java.base/java.net.PlainSocketImpl.socketConnect(Native Method) at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412) at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:255) at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237) at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) at java.base/java.net.Socket.connect(Socket.java:609) at org.postgresql.core.PGStream.createSocket(PGStream.java:243) at org.postgresql.core.PGStream.(PGStream.java:98) at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:132) at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:258) ... 41 more`

[Add relevant screenshots here if applicable]

## Issue at hand
https://github.com/orgs/I-TECH-UW/projects/6/views/1?pane=issue&itemId=57668194

[Add a link to the related issue or mention it here if applicable]

## Other
any sugestions are welcome @mozzy11 

[Add any additional information or notes here]

### CODE REVIEW

File: src/test/java/org/openelisglobal/DbUnitUtil.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
+import org.dbunit.DBTestCase;
+import org.dbunit.PropertiesBasedJdbcDatabaseTester;
+import org.dbunit.dataset.IDataSet;
+import org.dbunit.dataset.xml.FlatXmlDataSetBuilder;
+
+public class DbUnitUtil extends DBTestCase {
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/DbUnitUtil.java:12]: Why not extend `DataSourceBasedDBTestCase` which leverages a DataSource object for database connections. This is generally preferred over directly setting connection details as it allows for more flexibility and easier management of connections

Comment (mherman22) [reply src/test/java/org/openelisglobal/DbUnitUtil.java:12]: And then the error log you reported above, is it the same as what ci logs are reporting?

**Comment by valens200** [comment]
@josephbate   try this command and  see if there is any output : psql -U postgres.

File: src/test/java/org/openelisglobal/DbUnitUtil.java (hunk 1 above)
Comment (valens200) [reply src/test/java/org/openelisglobal/DbUnitUtil.java:12]: For more, you can take a look at: https://www.baeldung.com/java-dbunit

**Comment by josephbate** [comment]
> @josephbate try this command and see if there is any output : psql -U postgres.

@valens200 thank you for this ama try it out, though i wonder, does this test postgres in a test environment at runtime or in general ?
if it is the later then it may not serve the exact result, coz all the other tests that come with the clone repo work fine and are actually tested postgres db at runtime, Cc. @mozzy11 
the challenge i am facing is that despite all configurations made in the DdUnitUtil config class, it still fails to connect to the data base and store the data set iam trying to use    (for when tested at in momery the class works fine)

@valens200, i like how you are brain storming with ideas i believe we shall figure it out pretty soon thank you so much for the effort

**Comment by josephbate** [comment]
hello again @valens200 
the command you've suggested requires a preinstall postgres db locally, and to my knowledge the repo come with is own postgres db when using docker 
is there anythng i might be missing @valens200?

**Comment by sherrif10** [comment]
> knowledge the repo come with it's own postgres db when using docker
> is there anythng i might be missing

Are you able to connect to this db locally. ?

File: src/test/java/org/openelisglobal/person/PersonServiceTest.java [hunk 2]
```diff
@@ -23,10 +26,26 @@ public class PersonServiceTest {
...
+        dbUnitUtil = new DbUnitUtil("src/test/resources/dataSetFiles/personDataSet.Xml");
+
+        // Set up the database connection and load dataset using DbUnitUtil
+        IDatabaseTester databaseTester = new PropertiesBasedJdbcDatabaseTester();
+        databaseTester.setDataSet(dbUnitUtil.getDataSet());
+        databaseTester.onSetup();
     }
 
+     @Test
+    public void testingDataDataIsPopulated() {
```
Comment (sherrif10) [inline src/test/java/org/openelisglobal/person/PersonServiceTest.java:42]: i would suggesst` TestingDataIsPopulated`

**Comment by mozzy11** [comment]
Thanks @josephbate .
Not that we are currentlywotking on `develop `bramch not` develop_3x`

**Comment by mozzy11** [comment]
See merge conflicts

**Comment by josephbate** [comment]
> See merge conflicts

let me clear those 
meanwhile i have you taken a look on the errors is there any pointer you want to highlight

**Pull request closed without merging**